dev_scripts/generate-release-tasks.py

```diff
@@ -23,45 +29,29 @@ def git_root():
 
 
 def extract_checkboxes(filename):
-    headers = []
-    result = []
+    first_pass = []
+    second_pass = []
 
     with open(filename, "r") as f:
         lines = f.readlines()
 
-    current_level = 0
     for line in lines:
         line = line.rstrip()
+        if line.startswith("#") or line.lstrip().startswith("- [ ]"):
+            first_pass.append(line)
 
-        # If it's a header, store it
-        if line.startswith("#"):
-            # Count number of # to determine header level
-            level = len(line) - len(line.lstrip("#"))
-            if level < current_level or not current_level:
-                headers.extend(["", line, ""])
-                current_level = level
-            elif level > current_level:
-                continue
-            else:
-                headers = ["", line, ""]
-
-        # If it's a checkbox
-        elif "- [ ]" in line or "- [x]" in line or "- [X]" in line:
-            # Print the last header if we haven't already
-            if headers:
-                result.extend(headers)
-                headers = []
-                current_level = 0
+    current_level = 0
+    for line in reversed(first_pass):
+        level = len(line) - len(line.lstrip("#"))
+        if (level >= current_level > 0) or (level > 0 and not second_pass):
+            continue
+        current_level = level
+        second_pass.append(line)
```

Comment:
My previous code here seemed to be buggy in the past release, and I expect that will fix it, so thanks! I wonder if we should add some tests to ensure it's working as advertised? (not required, I'm genuinely wondering)

Reply:
I'd say it's minor really. It's a dev script for grouping release tasks for convenience, but we still have to read the release instructions.